target/scripts/helpers/log.sh

```diff
@@ -67,35 +66,41 @@ function _log() {
   case "${1}" in
     ( 'trace' )
       [[ ${LEVEL_AS_INT} -ge 5 ]] || return 0
-      MESSAGE+="  ${CYAN}TRACE  "
+      LOG_COLOR='CYAN'
+      LOG_LEVEL_NAME='TRACE'
       ;;
 
     ( 'debug' )
       [[ ${LEVEL_AS_INT} -ge 4 ]] || return 0
-      MESSAGE+="  ${PURPLE}DEBUG  "
+      LOG_COLOR='PURPLE'
+      LOG_LEVEL_NAME='DEBUG'
       ;;
 
     ( 'info' )
       [[ ${LEVEL_AS_INT} -ge 3 ]] || return 0
-      MESSAGE+="   ${BLUE}INF   "
+      LOG_COLOR='BLUE'
+      LOG_LEVEL_NAME='INFO '
       ;;
 
     ( 'warn' )
       [[ ${LEVEL_AS_INT} -ge 2 ]] || return 0
-      MESSAGE+=" ${LYELLOW}WARNING "
+      LOG_COLOR='LYELLOW'
+      LOG_LEVEL_NAME='WARN '
       ;;
 
     ( 'error' )
       [[ ${LEVEL_AS_INT} -ge 1 ]] || return 0
-      MESSAGE+="  ${LRED}ERROR  " ;;
+      LOG_COLOR='LRED'
+      LOG_LEVEL_NAME='ERROR'
+      ;;
 
     ( * )
       _log 'error' "Call to '_log' with invalid log level argument '${1}'"
       return 1
       ;;
   esac
 
-  MESSAGE+="${RESET}]  ${2}"
+  MESSAGE="$(date --iso-8601='seconds') ${!LOG_COLOR}${LOG_LEVEL_NAME}${RESET} $(basename "${0}"): ${2}"
```

Comment:
```suggestion
  MESSAGE="$(date --rfc-3339='seconds') ${!LOG_COLOR}${LOG_LEVEL_NAME}${RESET} $(basename "${0}"): ${2}"
```

Reply:
```
# date --iso-8601='seconds'
2024-05-27T22:00:56+02:00

# date --rfc-3339='seconds'
2024-05-27 22:01:18+02:00
```

Reply:
> ```console
> $ date --rfc-3339='seconds'
> 2024-05-27 22:01:18+02:00
> ```

Oh good catch! It should be able to support a `T` instead of space delimiter, but I guess `date` doesn't do that 🤷‍♂️ (_`logger` does as shown earlier, along with fractional secs at 6 digit precision_)

I just looked up RFC 5424 and it's actually requiring a specific format from RFC 3339 timestamps 😅

It seems RFC 3339 is usually chosen over ISO 8601 for [when software isn't intended to support the full range of ISO 8601 formats](https://www.reddit.com/r/ISO8601/comments/p572xy/rfc_3339_versus_iso_8601/), since most are only interested in the [intersection of the two standards](https://www.reddit.com/r/ISO8601/comments/v0903s/when_people_refer_to_the_iso8601_date_format_they/).

---

https://www.rfc-editor.org/rfc/rfc5424.html#section-6.2.3

> ![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/8a06b60e-6f3d-49fa-9d43-d90bbce2ae36)

> ![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/06ea7158-f220-46f2-b476-e1b8f5bd292f)

---

The [two standards differ a bit](https://ijmacd.github.io/rfc3339-iso8601/) but they both have the preferred format as `%Y-%M-%DT%h:%m:%s.%u%Z:%z` (_not a compatible `date` format_), the `date` command doesn't seem to support the fractional sec though.

The visual doesn't show the `:%z` suffix variant for UTC offset, but it's roughly equivalent to the highlighted one:

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/a31c42aa-12c3-423a-804d-ef77f9047429)

Reply:
> Oh good catch! It should be able to support a T instead of space delimiter

I liked the change, BECAUSE of the missing T 😆